File: FrontEnd/internal-app/src/app/core/models/table/sort.model.ts
/**
 * Sorting model
 * Use for config sorting on table
 */
export class SortModel {
  column: string;         // Column name (Name of column in database)
  direction: number;      // Direction
  isMultiOrder: boolean;  // Is multi sorting

  constructor(prop?: string, dir?: number, isMulti?: boolean) {
    this.column = prop || '';
    this.direction = dir || 0;
    this.isMultiOrder = isMulti || false;
  }
}

File: BackEnd/InternalApplication/Core/Core.Common/Models/SortModel.cs
namespace Core.Common.Models
{
    using Core.Common.Enums;
    using Newtonsoft.Json;

    /// <summary>
    /// Sort model.
    /// </summary>
    public class SortModel
    {
        /// <summary>
        /// Gets or sets property name of column.
        /// </summary>
        public string Column { get; set; }

        /// <summary>
        /// Gets or sets sort direction.
        /// </summary>
        public SortDirection Direction { get; set; }

        /// <summary>
        /// Gets or sets a value indicating whether allow order by multi column.
        /// </summary>
        public bool IsMultiOrder { get; set; } = false;

        /// <summary>
        /// Gets linq direction processing method.
        /// </summary>
        [JsonIgnore]
        public string LinqDirectionMethod
        {
            get
            {
                string method = null;
                if (this.Direction != SortDirection.None)
                {
                    switch (this.Direction)
                    {
                        case SortDirection.Ascending:
                            {
                                method = !this.IsMultiOrder ? "OrderBy" : "ThenBy";
                                break;
                            }

                        case SortDirection.Descending:
                            {
                                method = !this.IsMultiOrder ? "OrderByDescending" : "ThenByDescending";
                                break;
                            }
                    }
                }

                return method;
            }
        }
    }
}
